### ISSUE TITLE
Failure to parse a sequence of mapping values

### ISSUE THREAD

**Issue by agurney** [pulumi/kube2pulumi#65]
Some valid YAML input includes sequences of "mapping values" (each one is a single `k: v` pair) rather than a "mapping" of several of them. An example is a Kubernetes container's `envFrom` specification, for which see the example below. Currently, the intermediate `.pp` file is being generated with commas in the wrong places, which causes the pcl2pulumi step to fail with a parse error like

```
Error: Error: Missing item separator

  on pcl-640577668.pp line 20:
  15: envFrom = [
  16: {
  17: configMapRef = {
  18: name = "some-map"
  19: },
  20: }
  21: {

Expected a comma to mark the beginning of the next item.
```

This is a bit of an edge case in `yaml2pcl`, because it only manifests when the inner items are plain `k: v` pairs - for example, it _doesn't_ happen for inputs like

```yaml
  containers:
   - name: my-amazing-container
     image: something
     # ...
   - name: my-equally-amazing-sidecar
     image: something-else
     # ...
```
because here, each item in the sequence has more than one key.

## Expected behavior

Correct Pulumi code can be generated.

## Current behavior

It's not :-(

## Steps to reproduce

Run `kube2pulumi typescript -f sample.yaml` where `sample.yaml` has the contents below.
(This is just enough Kubernetes structure to show the troublesome bit, which is the two items under `envFrom`, not a realistic yaml fragment)

```yaml
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: thing
spec:
  replicas: 1
  template:
    spec:
      containers:
        - name: thing
          image: thing
          envFrom:
            - configMapRef:
                name: some-map
            - secretRef:
                name: some-secrets
```

## Context (Environment)

We specifically use this `envFrom:` structure, but there are probably other occurrences of the same pattern of types elsewhere in the wild.

## Affected feature

The relevant comma-insertion logic is in yaml2pcl when traversing a `SequenceNode`, 
https://github.com/pulumi/kube2pulumi/blob/master/pkg/yaml2pcl/yaml2pcl.go#L413 . Note that `suffix` will be either the empty string, or `","`, depending on whether we're on the last item in the list. The suffix is passed down to a recursive call to `walkToPCL`, which is hoped to do the right thing. 

In the case when the thing in the sequence is a `MappingValue`, its surrounding `{`/`}` are inserted on lines 423 and 440. Since line 440 comes _after_ the recursive call to `walkToPCL`, the comma (if any) is in the wrong place. The intermediate `.pp` file looks like this, with my comments added.

```
envFrom = [
{
configMapRef = {
name = "some-map"
},    # <-- comma added from the inner call to walkToPCL, line 408 of the code
}     # <-- comma not present, because line 440 does not add it
{
secretRef = {
name = "some-secrets"
}
}
]
```

Amending line 440 to
```go
				_, err = fmt.Fprintf(totalPCL, "%s%s\n", "}", suffix)
```
appears to produce the desired output in this case, but I haven't thoroughly tested that against other inputs.

In general, the handling of commas and delimiters is a bit hard to follow. I think there are other cases where incorrect results will come about, because not all code paths follow the same pattern. For example, in

```yaml
foo: ["bar", "baz"]
qux: [1, 2]
```

the `foo` items are handled properly, because the `ast.StringNode` case _does_ handle the suffix, but the `qux` ones are not, because `ast.IntegerNode` ignores the suffix. While the Kubernetes API doesn't include lists of integers (as far as I know), there may well be other bugs of this kind lurking.

**Comment by eblume** [comment]
I believe I may have identified a case where this prevents users from converting the official ArgoCD install.yaml manifest using `kube2pulumi`. I discussed this in this thread on the community slack: https://pulumi-community.slack.com/archives/CDE799L1M/p1676930179533059?thread_ts=1676928416.259339&cid=CDE799L1M

The manifest can be found here: https://raw.githubusercontent.com/argoproj/argo-cd/master/manifests/install.yaml

The relevant section of the manifest is:

```yaml
        ports:
        - containerPort: 5556
        - containerPort: 5557
        - containerPort: 5558
```

The error output is:

```
unable to run program: Error: Missing item separator

  on pcl-1017078102.pp line 1301:
1298: ports = [
1299: {
1300: containerPort = 5556
1301: }
1302: {

Expected a comma to mark the beginning of the next item.
```
